exes/Main.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE PatternSynonyms #-}
```

Comment:
Shouldn't this rather be added to the `default-extensions` in the `common defaults` section of the `.cabal` file?
It is also worth considering to switch to `default-language: GHC2021` because we only support GHC 9.6 and up now.

Reply:
`PatternSynonyms` is an essential part of the Haskell language, imo, as it provides a robust syntax for importing constructors (as in `import M (pattern c)`), so I think it should be _always on_.

But maybe this can be taken in another PR.

Reply:
```
    The ‘pattern’ namespace specifier is deprecated.
    Suggested fix:
      Use the ‘ExplicitNamespaces’ extension
```

Reply:
Thanks @phadej, for the heads up!  
The `data` namespace is being added in 9.14:
> data is enabled by [ExplicitNamespaces](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/explicit_namespaces.html#extension-ExplicitNamespaces) and requires GHC ≥9.14